cargo/cargo_build_script.bzl

```
@@ -32,13 +33,21 @@ def _cargo_build_script_run(ctx, script):
     env = {
         "CARGO_CFG_TARGET_ARCH": toolchain.target_arch,
         "CARGO_MANIFEST_DIR": manifest_dir,
+        "CARGO_PKG_NAME": crate_name,
         "HOST": toolchain.exec_triple,
         "OPT_LEVEL": compilation_mode_opt_level,
         "RUSTC": toolchain.rustc.path,
         "TARGET": toolchain.target_triple,
         # OUT_DIR is set by the runner itself, rather than on the action.
     }
 
+    if ctx.attr.version:
+        version = ctx.attr.version.split(".")
+        env["CARGO_PKG_VERSION"] = ctx.attr.version
+        env["CARGO_PKG_VERSION_MAJOR"] = version[0]
+        env["CARGO_PKG_VERSION_MINOR"] = version[1] if len(version) > 1 else ""
+        env["CARGO_PKG_VERSION_PATCH"] = version[2] if len(version) > 2 else ""
```

What about when there is a label (e.g. `1.2.3-blah` or `2.3.4+bleh`)?

```suggestion
    if ctx.attr.version:
        version = ctx.attr.version.split("+")[0].split("-")[0].split(".")
        env["CARGO_PKG_VERSION"] = ctx.attr.version
        env["CARGO_PKG_VERSION_MAJOR"] = version[0]
        env["CARGO_PKG_VERSION_MINOR"] = version[1] if len(version) > 1 else ""
        env["CARGO_PKG_VERSION_PATCH"] = version[2] if len(version) > 2 else ""
```

---

done. slightly different though.